src/Essentials/src/Geolocation/Geolocation.shared.cs

```diff
@@ -27,6 +29,40 @@ public interface IGeolocation
 		/// <remarks>The location permissions will be requested at runtime if needed. You might still need to declare something in your app manifest.</remarks>
 		/// <returns>A <see cref="Location"/> object containing current location information or <see langword="null"/> if no location could be determined.</returns>
 		Task<Location?> GetLocationAsync(GeolocationRequest request, CancellationToken cancelToken);
+
+		/// <summary>
+		/// Indicates if currently listening to location updates while the app is in foreground.
+		/// </summary>
+		bool IsListeningForeground { get; }
+
+		/// <summary>
+		/// Occurs while listening to location updates.
+		/// </summary>
+		event EventHandler<LocationEventArgs>? LocationChanged;
+
+		/// <summary>
+		/// Occurs when an error during listening for location updates arises. When getting the event,
+		/// listening for further updates may have been stopped.
+		/// </summary>
+		event EventHandler<GeolocationErrorEventArgs>? LocationError;
```

Comment:
I think .NET worlds use "failed"? So maybe 

```cs
event EventHandler<GeolocationListeningFailedEventArgs>? ListeningFailed;
```

If this works more with the listening things, then we should have listening. People might expect this to fire if any location things fail - like a normal GetLocationAsync.

Reply:
I see also the listen will throw an exception if something prevents the listener starting. So I am seeing that this event will only fire if something causes a problem _while_ a listening is happening - ie: permissions revoked.

I feel like we need "listening" in the name even more now.

Reply:
I renamed the event to `ListeningFailed`and the class to `GeolocationListeningFailedEventArgs`.